cider/exec/template/DateTimeIR.cpp

```diff
@@ -70,12 +70,40 @@ const char* get_extract_function_name(ExtractField field) {
     case kYEAR:
       return "extract_year";
   }
-  UNREACHABLE();
+  CIDER_THROW(CiderUnsupportedException, fmt::format("field is {}", field));
   return "";
 }
 
+bool is_extract_function_complex(ExtractField field) {
```

Comment:
please define the ExtractField's enum with value like this:
```c++
enum ExtractField {
  kYEAR = 0x00000001,
  kQUARTER  = 0x00000002,
  kMONTH = 0x00000004,
  kDAY = 0x00000008,
  kHOUR = 0x00000010,
  kMINUTE = 0x00000020,
  kSECOND = 0x00000040,
  kMILLISECOND = 0x00000080,
  ...
};
```
and for following pattern match, we can use bit operation to make faster